UserTools/VertexGeometryCheck/VertexGeometryCheck.cpp

```diff
@@ -31,150 +35,192 @@ bool VertexGeometryCheck::Initialise(std::string configfile, DataModel &data){
+        // Y vs theta
+        double theta = 0.0;
+        double thetadeg = 0.0;
+        if (digitZ != 0.0) {
+            theta = atan(digitX / digitZ);
+        }
+        if (digitZ <= 0.0) {
+            if (digitX > 0.0) theta += TMath::Pi();
+            if (digitX < 0.0) theta -= TMath::Pi();
+        }
+        thetadeg = theta / (TMath::Pi() / 180.0); // radians->degrees   
+        if(fDigitList->at(n).GetDigitType() == RecoDigit::PMT8inch)fYvsDigitTheta_all->Fill(thetadeg, digitY, digitT);
+        if(fDigitList->at(n).GetDigitType() == RecoDigit::lappd_v0)fYvsDigitTheta_all->Fill(thetadeg, digitY, digitT);
+        fdelta->Fill(myvtxgeo->GetDelta(n));
+        fpointtres->Fill(myvtxgeo->GetPointResidual(n));
+        if (myvtxgeo->GetDigitType(n) == RecoDigit::lappd_v0) flappdextendedtres->Fill(myvtxgeo->GetExtendedResidual(n));
+        if (myvtxgeo->GetDigitType(n) == RecoDigit::PMT8inch) fpmtextendedtres->Fill(myvtxgeo->GetExtendedResidual(n));
+        fltrack->Fill(myvtxgeo->GetDistTrack(n)); //cm
+        flphoton->Fill(myvtxgeo->GetDistPhoton(n)); //cm
+        fzenith->Fill(phideg/*myvtxgeo->GetZenith(n)*/); //
+        fazimuth->Fill(myvtxgeo->GetAzimuth(n)); //
+        fconeangle->Fill(myvtxgeo->GetConeAngle(n)); //
+        fdigitcharge->Fill(myvtxgeo->GetDigitQ(n));
+        fdigittime->Fill(digitT);
+        if (myvtxgeo->GetDigitType(n) == RecoDigit::lappd_v0) flappdtimesmear->Fill(Parameters::TimeResolution(RecoDigit::lappd_v0, myvtxgeo->GetDigitQ(n)));
+        if (myvtxgeo->GetDigitType(n) == RecoDigit::PMT8inch) fpmttimesmear->Fill(Parameters::TimeResolution(RecoDigit::PMT8inch, myvtxgeo->GetDigitQ(n)));
+        std::cout << "VGCheck " << n << endl;
+        if (StripTimePlot > 0) {
+            if (fDigitList->at(n).GetDigitType() == RecoDigit::lappd_v0 && fDigitList->at(n).GetCalTime()>0 /* && fDigitList->at(n).GetDetectorID() == StripTimePlot*/) {
+                if (fDigitList->at(n).GetDetectorID() != currentLAPPD)
+                {
+
+                    if (currentLAPPD > 0 && iStripHit > 0) {
+                        stripPlotName = "StripPlot_e" + std::to_string(fEventNumber) + "lappd" + std::to_string(currentLAPPD);
+                        fOutput_tfile->cd();
+                        StripHits1->Write(stripPlotName.c_str());
+                    }
+                    StripHits1->Reset();
+                    currentLAPPD = fDigitList->at(n).GetDetectorID();
+                    iStripHit = 0;
+                }
+                StripHits1->Fill(fDigitList->at(n).GetPosition().X(), fDigitList->at(n).GetCalTime() * 1000/*, fDigitList->at(n).GetCalCharge()*/);
+                iStripHit++;
+                std::cout << "VGCheck striphit";
+                
+            }
+        }
+    }
+
+    delete myFoMCalculator;
+
   return true;
 }
 
 bool VertexGeometryCheck::Finalise(){
   fOutput_tfile->cd();
   fOutput_tfile->Write();
+  if (StripTimePlot > 0) StripHits1->Write("lappdtimegradient");
```

Comment:
this name doesn't seem consistent with the usage in the above loop. Is this correct?